### ISSUE TITLE
Driller Anchor in first hole repeats and error Z

### ISSUE THREAD

**Issue by MARIOBASZ** [vlachoudis/bCNC#832]
The first hole is done at the wrong height (random?) (Can you tell me why I learn so?) Then do it correctly and repeat it one last time if it is a closed route

![image](https://user-images.githubusercontent.com/11743403/39948873-4dc5f348-554e-11e8-968e-52a6648d6058.png)

It is a triangle, there should be 3 holes, there are 5 and the first is at Z = 6, when it should be z = -1
I add a pull request

**Comment by buschhardt** [comment]
See https://github.com/vlachoudis/bCNC/pull/833 maybe it helps with closed path.

**Comment by MARIOBASZ** [comment]
I found this error: if there is more than one block and you choose one different from the first one, it generates a first additional hole (I think there is a function to eliminate the points external to the route)

**Comment by MARIOBASZ** [comment]
Can you guide me where to find list (OrderedDict.fromkeys (bidHoles) to understand how it works to clear duplicates?

**Comment by MARIOBASZ** [comment]
A minor improvement could be to eliminate duplicates in xy plane (it does not make sense to make more than one hole if the movement is just change of height without translation)

**Comment by buschhardt** [comment]
> Can you guide me where to find list (OrderedDict.fromkeys (bidHoles) to understand how it works to clear duplicates?

Yes, the OrderedDict(ionary)-Class is a special Hash that have also a reference to an array. In the normal case Dictionaries are unordered - you looking only for the key and get the value - it dosnt matter where the key is. But if you need the original order, then you can use this class.
How it works here:
1. take all listmembers as keys of this class [p1,p2,p3...]->{p1:null,p2:null,p3:null}
2. if the import occur there is only one key with the same name possible [p1,p1,p2]->{p1:null,p2:null}
3. fromkeys -> we interest only in keys
4. make a new list from that keys
Keys can also be other objects (without Dictionaries, I believe)
```
from collections import OrderedDict
bidHoles = list(OrderedDict.fromkeys(bidHoles))
```

> A minor improvement could be to eliminate duplicates in the xy plane (it does not make sense to make more than one hole if the movement is only height change without moving)

If You use the driller plugin - I eleminate only the duplicates from a block - not from all blocks. Should we?

**Comment by MARIOBASZ** [comment]
Thanks
Suppose the previous block ends at a low height, Houstn we have a problem!
you just punched the neighbors below!

independently that the previous block is unmarked
GCode in .TXT

[PRUEBA ANCHOR DRILL.TXT](https://github.com/vlachoudis/bCNC/files/1997660/PRUEBA.ANCHOR.DRILL.TXT)


![image](https://user-images.githubusercontent.com/11743403/39957949-e57a179a-55d1-11e8-9143-90df78b4e0a0.png)

**Comment by buschhardt** [comment]
I cant find an error, please describe every step to reproduce Your procedure. The textfile inst that from the screenshot.

**Comment by buschhardt** [comment]
OK I see if I have 2 blocks selected and run the plugin, it will add the resulted gcode in one block.
But this is not an error, or?

**Comment by MARIOBASZ** [comment]
I do not know why it is comparing with version 0.0.8 instead of 0.0.10
Below I clarified the few modified lines

**Comment by MARIOBASZ** [comment]
I have seen where the problem, in anchor, continues to appear in the segments originated with G0 or Zonli
